src/pkg/backup/details/details.go

```diff
@@ -305,6 +305,84 @@ func (d *Details) addFolder(folder folderEntry) {
 	})
 }
 
+// FilterMetaFiles returns a new Details struct with a copy of the DetailsModel
+// that has had all .meta and .dirmeta files stripped out.
+func (d *Details) FilterMetaFiles() *Details {
+	return &Details{
+		DetailsModel: d.DetailsModel.FilterMetaFiles(),
+	}
+}
+
+// FilterEmptyContainers returns a new Details struct all empty (ie: containing no
+// items) stripped out.  If meta files have not been filtered out already, they
+// will continue to count as a "populated" container.
+func (d *Details) FilterEmptyContainers() *Details {
```

Comment:
Was trying to see if there's a better way to do this instead of going through all folders/items again. We're already scanning this data structure multiple times.

Where are empty folders actually being added to details? Should we just stop doing that till we have a solution to handle restore of empty directories?

Reply:
> Where are empty folders actually being added to details? Should we just stop doing that till we have a solution to handle restore of empty directories?

If we can swallow the performance cost in the short term, this will also act as a good location to feature-flag the progression towards backing up and restoring empty folders.  Preventing their upload in the first place is effectively just a rollback on Ash's prior work, and a step back from our eventual goal.

Reply:
I might be overthinking it - can you push a small bench test that checks the overhead of running the two filter operations on a details model with 300K items. I can run it on some of the test systems I'm using for benchmarking (not our MacBook Pro's)

Reply:
>Where are empty folders actually being added to details? Should we just stop doing that till we have a solution to handle restore of empty directories?

Mostly these are coming from OneDrive as empty Exchange folders shouldn't create details entries. For OneDrive it's much easier to handle things if we don't exclude empty folders due to the fact that we see folders before items and a weird interaction between selector filtering and Graph API item ordering. I would expect to only see empty folders if backup permissions was enabled in OneDrive though. Otherwise empty folders may be made in kopia but there shouldn't be any items in backup details that cause a folder entry to exist in backup details for that folder. We're seeing empty folders purely because we filter some things out of backup details

Reply:
actually, I stand corrected, corso will save empty permissions files when backup permissions is not enabled. That would result in empty folders being present in the backup details